hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -425,6 +429,30 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 			}
 		}
 
+		// Once the deletion has occurred, we need to clean up cluster-wide resources
+		selector := client.MatchingLabelsSelector{Selector: labels.SelectorFromSet(labels.Set{
+			controlplanepkioperatormanifests.OwningHostedClusterNamespaceLabel: hcluster.Namespace,
+			controlplanepkioperatormanifests.OwningHostedClusterNameLabel:      hcluster.Name,
+		})}
+		var crs rbacv1.ClusterRoleList
+		if err := r.List(ctx, &crs, selector); err != nil {
+			return ctrl.Result{}, fmt.Errorf("failed to list cluster roles: %w", err)
+		}
+		if len(crs.Items) > 0 {
+			if err := r.DeleteAllOf(ctx, &crs.Items[0], selector); err != nil {
```

Comment:
nit: It would be clearer to me if you use &rbacv1.ClusterRole{} instead of &crs.Items[0], given that what you need to pass is the type and not necessarily any part of the list